consensus/hotstuff/verification/combined_verifier_v2.go

```diff
@@ -147,17 +148,21 @@ func (c *CombinedVerifier) VerifyQC(signers flow.IdentityList, sigData []byte, b
 	// TODO: update to use module/signature.PublicKeyAggregator
 	aggregatedKey, err := crypto.AggregateBLSPublicKeys(signers.PublicStakingKeys()) // caution: requires non-empty slice of keys!
 	if err != nil {
-		// `AggregateBLSPublicKeys` returns a `crypto.invalidInputsError` in two distinct cases:
+		// `AggregateBLSPublicKeys` returns an error in two distinct cases:
 		//  (i) In case no keys are provided, i.e.  `len(signers) == 0`.
 		//      This scenario _is expected_ during normal operations, because a byzantine
 		//      proposer might construct an (invalid) QC with an empty list of signers.
 		// (ii) In case some provided public keys type is not BLS.
 		//      This scenario is _not expected_ during normal operations, because all keys are
 		//      guaranteed by the protocol to be BLS keys.
-		//
-		// By checking `len(signers) == 0` upfront , we can rule out case (i) as a source of error.
-		// Hence, if we encounter an error here, we know it is case (ii). Thereby, we can clearly
-		// distinguish a faulty _external_ input from an _internal_ uncovered edge-case.
+		// check case (i)
+		// TODO: this should not happen because the case `len(signers) == 0`
+		// is checked upfront and ruled out. This is because `Unpack` does not
+		// handle the empty list case properly.
```

Comment:
> Do you want to indicate that in the future we should make the following revisions?

Yes, that was the meaning of the TODO. 

My bad, I initially thought the tests were failing because of `Unpack`, but they were failing because of mockery issues. I did fix the mocks in tests but didn't get back to testing `Unpack` with an empty list. 

Thank you for cleaning the checks in https://github.com/onflow/flow-go/pull/3677. The TODOs are now not needed.

Reply:
I actually dug further, and it turns out the empty list case was failing because the test itself. `Unpack` was failing because the test was not built properly I think. Fixed in [e5e3930](https://github.com/onflow/flow-go/pull/3441/commits/e5e3930ff0f9c2456829b221c95c2af821bcfc3f)